addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -693,6 +725,53 @@ var SnippetEditor = Widget.extend({
             $optionsSection.toggleClass('d-none', options.length === 0);
         });
     },
+    /**
+     * Manages the overlay visibility towards animations/transitions
+     * states. This prevents having the overlay lagging behind the snippet
+     * css animation (with animation previews from 'website_animate' among others).
+     *
+     * @private
+     * @param {number} [animationsCountDelta=0] The delta to apply on the
+     * animations count as integer.
+     * @param {boolean} [animationTimeoutCall=false] is the call to the function made from
+     * the animation timeout function?
+     */
+    _manageOverlayMoves: function (animationsCountDelta = 0) {
+        const maxDuration = this._getTargetMaxAnimationAndTransitionDuration();
+        this._overlayMovesCounters += animationsCountDelta;
+        this._overlayMovesCounters = Math.max(0, this._overlayMovesCounters);
+        clearTimeout(this._postAnimationHandler);
+        clearTimeout(this._animationTimeoutHandler);
+        if (this._overlayMovesCounters > 0) {
+            this._toggleAllOverlaysVisibility(false);
+            // The normal case would be to have the transitionend event to be
+            // fired but we cannot rely on it, so we use a timeout as fallback.
+            this._animationTimeoutHandler = setTimeout(
+                () => {
+                    this._manageOverlayMoves(-this._overlayMovesCounters);
+                },
+                Math.max(maxDuration + 100, 1000)
+            );
+        } else {
+            // Using a Timeout here prevents the cancel animation events
+            // from occurring when changing rapidly the animation (which triggered overlay flicker).
+            this._postAnimationHandler = setTimeout(
+                () => {
+                    // Force calling cover() on all overlays by triggering resize on the window.
+                    $(window).trigger("resize");
+                    this._toggleAllOverlaysVisibility(true);
+                },
+                100
+            );
+        }
+    },
+    /**
+     * @private
+     * @param {boolean} [visible]
+     */
+    _toggleAllOverlaysVisibility: function (visible) {
+        this.$allOverlaysParent.toggleClass('overlay_hidden_during_animation', !visible);
+    },
```

Comment:
Instead of create a new function you can use toggleTextEdition() already there. And change the name of it => toggleOverlayVisibility(). And replace the o_keypress class by o_overlay_hidden.
You can check in this commit (not merged yet) how I did it.. So it will be easier to merge ;)
https://github.com/odoo/odoo/pull/51446/commits/1da6fc3537cec73ea51ca6841e8b0d018f32b05f

Reply:
No really I think.

The problem is that it could be possible that an animation is triggered and still running while another overlay is displayed (as you hover its config area on the left). In that case the overlay should not be displayed and the only solution I found is to prevent any overlay to be displayed as soon as an animation is trigged until none are running.

Trying to find all the overlays of current overlay's target children's overlays would be overkill I think. I hope what I just wrote is understandable :o)